app/routes/registration/form.js

```diff
@@ -0,0 +1,28 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+    model() {
+        return this.store.createRecord('registration', { directSource: false });
+    },
+    actions: {
+        saveRegistrationModel(model) {
+            var registration = model;
+            var _this = this;
+
+            if (registration.validate()) {
+                registration.save().then(
+                    function() {
```

Comment:
Avoid using  `this` aliases, instead use [lexical this/fat arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).